src/main/java/com/mitchej123/hodgepodge/core/handlers/ClientKeyListener.java

```diff
@@ -16,12 +17,14 @@ public class ClientKeyListener {
     public void keyUp(InputEvent.KeyInputEvent event) {
         int key = Keyboard.getEventKey();
         boolean released = !Keyboard.getEventKeyState();
-        if (Minecraft.getMinecraft().gameSettings.showDebugInfo
-                && GuiScreen.isShiftKeyDown()
-                && key == Keyboard.KEY_N
-                && released) {
-            HodgePodgeClient.animationsMode =
-                    HodgePodgeClient.animationsMode == 2 ? 0 : ++HodgePodgeClient.animationsMode;
+        if (Minecraft.getMinecraft().gameSettings.showDebugInfo && GuiScreen.isShiftKeyDown() && released) {
+            if (key == Keyboard.KEY_N) {
+                HodgePodgeClient.animationsMode =
+                        HodgePodgeClient.animationsMode == 2 ? 0 : ++HodgePodgeClient.animationsMode;
```

Comment:
That was a quick dirty implementation and this should be implemented in a much cleaner way with a state manager of some kind.

Reply:
I'm not sure what kind of manager you had in mind, but I hope this is enough.